backend/Repositories/WhaleRepo.cs

```diff
@@ -18,11 +19,20 @@ public WhaleRepo(WhaleSpottingDbContext context)
         {
             _context = context;
         }
-         public IEnumerable<Species> GetAllSpecies()
+        public IEnumerable<Species> GetAllSpecies()
         {
             return _context.Species
                 .Include(s => s.ConservationStatus)
                 .OrderBy(s => s.Name);
         }
+
+        public Species GetSpeciesById(int speciesId) 
+        {
+            return _context.Species
+                .Include(s => s.ConservationStatus)
+                .Where(s => s.Id == speciesId)
+                .Single();
```

Comment:
The `Single()` method is quite cool - you can pass it the predicate and it does the filtering for you!
```suggestion
                .Single(s => s.Id == speciesId);
```

Reply:
done, thanks